### DISCUSSION TITLE
Take ownership in meta

### DISCUSSION THREAD

**Discussion by mathisloge** [skypjack/entt#1177] [Q&A]
Hey, 
is there a way to take the ownership of the underlying data if the `meta_any` was created with `from_void` so that it is deleted if it goes out of scope?

**Comment by skypjack** [comment]
No, there isn't a way to do it (yet?). I think it's possible to instruct a `meta_any` to get the ownership of the element but it's not available on the current interface actually. What would be the use case though? From where you get this `void *` that you can steal it and release it freely? I'm curious now. 🙂

**Reply by mathisloge** [reply]
I'm currently developing a remote ui testing framework for Qt. And if I'm invoking a meta function of qt, I have to allocate the exact return type of the function. And for that the qt meta system is giving me back a void pointer which I later have to destroy manually.

My wrapper to call such a meta function returns an entt::meta_any which now holds the void pointer.

Currently the function returns a structure which holds the ownership in form of a unique pointer and a custom deleter and the meta_any.

But I've wondered if it would be possible to directly use the entt meta any type to take the ownership.

**Reply by skypjack** [reply]
I think this is an interesting call an maybe also a viable change. I'm creating an issue from the discussion to investigate it further. 👍
